obs-check-package-origin

```diff
@@ -0,0 +1,156 @@
+#!/bin/bash -e
+
+# Motivation: https://progress.opensuse.org/issues/154723
+# 1. fetch requirements from package spec file,
+# 2. look for package names for such requirements
+# 3. search all relevant instances where the package is maintained
+# 4. fetch version info from spec file and from Tumbleweed repo
+
+osc="${osc:-"osc --apiurl https://api.opensuse.org"}"
+
+BLUE="\e[94m"
+RED="\e[91m"
+GRAY="\e[90m"
+ENDCOLOR="\e[0m"
+
+debug() {
+    test -n "$DEBUG" && echo -e "${GRAY}$1${ENDCOLOR}" >&2
+}
+
+info() {
+    echo -e "${BLUE}$1${ENDCOLOR}" >&2
+}
+
+error() {
+    echo -e "${RED}$1${ENDCOLOR}" >&2
+}
+
+init_check() {
+    for cmd in which yq rpmspec osc zypper; do
+        which "$cmd" &> /dev/null || {
+            error "$cmd command is missing"
+            exit 100
+        }
+    done
+}
+
+parse_zypper() {
+    yq -pxml '.stream.search-result.solvable-list | (.solvable.+@name?,.solvable[].+@name)' \
+        | grep -v null \
+        | sort -u \
+        | tr '\n' ' '
+}
+
+get_package_version() {
+    local spec
+    spec="$(basename "$1").spec"
+    local tmpfile
+    tmpfile="$(mktemp)"
+    debug "Getting version from spec file for '$1'"
+    $osc cat "$1" "$spec" > "$tmpfile"
+    rpmspec -q --qf '%{version}\n' "$tmpfile" 2> /dev/null | head -n1
+    rm "$tmpfile"
+}
+
+get_tw_version() {
+    local package
+    package="$(basename "$1")"
+    test -n "$ZYPPER" || return
+    debug "Getting Tumbleweed version for '$package'"
+    zypper -n --no-refresh info "$package" | grep ^Version | cut -f2 -d: | tr -d ' '
+}
+
+get_package_name() {
+    local req
+    req="$(basename "$1" | cut -f1 -d' ')"
+    debug "Looking for package name for requirement '$req'"
+    $osc se --package "$req" | grep -oP "'[^']+'" | head -n1 | tr -d "'"
+}
+
+list_requirements() {
+    local package
+    local tmpfile
+    debug "Getting build requirements from '$1.spec'"
+    tmpfile="$(mktemp)"
+    $osc cat devel:openQA "$1" _service:obs_scm:"$1.spec" > "$tmpfile"
+    rpmspec -q -D "sysusers_requires BuildRequires: sysuser-tools" --buildrequires "$tmpfile" \
+        | sort -u \
+        | while read -r; do
+            package="$(get_package_name "$REPLY")"
+            test -n "$package" || {
+                error "Unable to get package name for '$REPLY'"
+                continue
+            }
+            echo "$package"
+        done
+    rm "$tmpfile"
+}
```

Comment:
@baierjan maybe you want to look at how I did it at https://progress.opensuse.org/issues/154747#note-3

Not entirely sure if buildrequires will also pull requires (the distri doesn't have buildrequires for instance)